x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/inference/persistence/TrainedModelProvider.java

```diff
@@ -208,7 +215,7 @@ public void storeTrainedModelConfig(TrainedModelConfig trainedModelConfig, Actio
             ML_ORIGIN,
             TransportIndexAction.TYPE,
             request,
-            ActionListener.wrap(indexResponse -> listener.onResponse(true), e -> {
+            ActionListener.wrap(indexResponse -> refreshCacheVersion(listener), e -> {
```

Comment:
`TrainedModelProvider` is used to persist and read the config for both the Tree ensemble and PyTorch models we support.  PyTorch models are not cached so we can save a cluster state update by only refreshing the cache only if 
`trainedModelConfig.getModelType() == TrainedModelType.TREE_ENSEMBLE`.

Reply:
My preference goes for updating the cache metadata consistently regardless of the model type. 
It provides a way to detect model changes if someone needs to implement such a thing for pytorch models.

Additionally, those are small data and they are not updated very often.